logic/tool/TypeHinter.java

```diff
@@ -221,12 +214,22 @@ private void addInferredSubLabels(TypeVariable variable, Set<Label> hints, Map<L
     }
 
     private void addInferredIsaLabels(ThingVariable variable, Set<Label> hints, Map<Label, TypeVariable> labelMap) {
+        removeAbstractHints(hints);
         if (!variable.isa().isPresent()) {
             variable.isa(lowestCommonSuperType(hints, labelMap), false);
         }
         variable.isa().get().addHints(hints);
```

Comment:
is there an `else` missing here??

Reply:
No. This code checks if an `isa` edge exists, and if not adds it. If it already exists, we just carry on.

Reply:
cool!